torch/testing/_internal/common_device_type.py

```diff
@@ -570,8 +560,9 @@ def inner(fn):
         @wraps(fn)
         def dep_fn(self, *args, **kwargs):
             size_bytes = size(self, *args, **kwargs) if callable(size) else size
-            if not _has_sufficient_memory(self.device_type, size_bytes):
-                raise unittest.SkipTest('Insufficient {} memory'.format(self.device_type))
+            _device = self.device_type if device is None else device
```

Comment:
The device can't be acquired from the device_type here because the device_type would be 'cuda' but the actual device might be 'cuda:1'. The decorator needs to take a device_type in the decorator, like 'cuda', but then the check here needs to acquire the (primary) device. See:

https://github.com/pytorch/pytorch/blob/6de619e4a4fd7ce55c3a2f86e666fa9f29026724/torch/testing/_internal/common_device_type.py#L335